scripts/coredump/coredump_serial_log_parser.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2020 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import binascii
+import sys
+
+
+COREDUMP_PREFIX_STR = "#CD:"
+
+COREDUMP_BEGIN_STR = COREDUMP_PREFIX_STR + "BEGIN#"
+COREDUMP_END_STR = COREDUMP_PREFIX_STR + "END#"
+COREDUMP_ERROR_STR = COREDUMP_PREFIX_STR + "ERROR CANNOT DUMP#"
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument("infile", help="Serial Log File")
+    parser.add_argument("outfile",
+            help="Output file for use with coredump GDB server")
+
+    return parser.parse_args()
+
+
+def main():
+    args = parse_args()
+
+    infile = open(args.infile, "r")
+    if not infile:
+        print(f"ERROR: Cannot open input file: {args.infile}, exiting...")
+        sys.exit(1)
+
+    outfile = open(args.outfile, "wb")
+    if not outfile:
+        print(f"ERROR: Cannot open output file for write: {args.outfile}, exiting...")
+        sys.exit(1)
+
+    print(f"Input file {args.infile}")
+    print(f"Output file {args.outfile}")
+
+    has_begin = False
+    has_end = False
+    has_error = False
+    go_parse_line = False
+    bytes_written = 0
+    for line in infile.readlines():
+        if line.find(COREDUMP_BEGIN_STR) >= 0:
```

Comment:
shouldn't `go_parse_line` be initialized to `False` in the beginning of this loop ? Otherwise you may have a situation where the previous line was `COREDUMP_BEGIN_STR`, consequently `go_parse_line` is `True` now, and the current line does not trigger any of these `if` causing the line to be invalid parsed.

Is that a possible scenario ?

Reply:
This is mainly to skip all the lines before beginning of coredump log, since log is enclosed with `BEGIN`/`END` line pair. In other words, this ensures only lines are parsed after `BEGIN` has been seen (and until `END`).